lib/Makefile.am

```diff
@@ -43,7 +43,7 @@
 # Library version info - here at the top, for sanity
 # See <http://www.gnu.org/software/libtool/manual/libtool.html#Versioning>
 # CURRENT:REVISION:AGE
-sasl_version = 3:0:0
+sasl_version = 4:0:1
```

Comment:
I think AGE should be kept at 0 since the layout of `struct sasl_utils` is changed incompatibly and I can't see any other way the library tries to maintain support for plugins linked against `libsasl2.so.3` either.

Reply:
Quoting https://www.gnu.org/software/libtool/manual/libtool.html#Versioning :

     If any interfaces have been added, removed, or changed since the last update, increment current, and set revision to 0.
     If any interfaces have been added since the last public release, then increment age.

Reply:
My reading is that the `struct sasl_utils` part of the interface is "changed" (a plugin populating or accessing that struct will not work with new ABI) so we can't support v3 and v4 simultaneously (which we claim when setting AGE to 1).

Reply:
An old plugin will work with a new library. The added fields just occupy the "spare" fields in the old structure so old plugins would ignore them.

Reply:
I don't understand, it looks like the structure has just grown by 2 pointers in the middle (shifting everything other member down a little), I can't square that with your suggestion.

Reply:
Then those spare pointers should be considered replaced by this, not added, so as not to affect the struct's size?

Reply:
Irrelevant. The struct is allocated by libsasl, plugins don't know or care about its size.

Reply:
To sum up the above and an out-of-band conversation:

`struct sasl_utils` is meant to be set up by liblsasl and considered read-only by plugins, so anything past its end is not considered part of the ABI. Adding these, we have to tell new users that the members are available (moving our soname to liblsasl2.so.4) but we haven't changed the meaning of anything in the v3 ABI and anyone linked against liblsasl2.so.3 should work just as it did before (and we can bump AGE to 1 to say we support both interfaces).

Given we can do the same next time something like this is needed, it turns out those extra pointers were never necessary and being unused, we can and will drop them. This might also have the side effect of helping automated ABI checkers understand the intent of such changes.